src/ml_tooling/data/base_data.py

```diff
@@ -0,0 +1,84 @@
+import abc
+from typing import Optional, Tuple
+
+from sklearn.model_selection import train_test_split
+from ml_tooling.utils import DataType, DataSetError
+from sklearn.utils import indexable
+
+
+class DataSet(metaclass=abc.ABCMeta):
+    _x: Optional[DataType] = None
+    _y: Optional[DataType] = None
+    test_x: Optional[DataType] = None
+    test_y: Optional[DataType] = None
+    train_y: Optional[DataType] = None
+    train_x: Optional[DataType] = None
+
+    def create_train_test(
+        self, stratify=None, shuffle=True, test_size=0.25, seed=42
+    ) -> "DataSet":
+        """
+        Creates a training and testing dataset and storing it on the data object.
+        :param stratify:
+            What to stratify the split on. Usually y if given a classification problem
+        :param shuffle:
+            Whether or not to shuffle the data
+        :param test_size:
+            What percentage of the data will be part of the test set
+         :param seed:
+            Random seed for train_test_split
+        :return:
+            self
+        """
+
+        self.train_x, self.test_x, self.train_y, self.test_y = train_test_split(
```

Comment:
Here we assume that we always want to use a simple train_test_split. Maybe we should create a wrapper for a Shuffler where the default used Shuffler is ShuffleSplit. That way one could also easliy use GroupShuffleSplit, TimeSeriesSplit......

Reply:
I would probably do that as part of the implementation of internal cross-validation splitting - then this would be calling that with CV=1

Reply:
I have created an issue we can work on later then.